### PR TITLE
correctly wiring up OnChange for ScriptApplicationHostOptions

### PR DESCRIPTION
**Pull request by brettsam**

### CODE REVIEW

File: test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SpecializationE2ETests.cs [hunk 1]
```diff
@@ -246,6 +243,51 @@ public async Task Specialization_ResetsSharedLoadContext()
             }
         }
 
+        [Fact]
+        public async Task Specialization_ResetsSecretManagerRepository()
```
Comment (liliankasem) [inline test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SpecializationE2ETests.cs:247]: Are there any edge cases we can test for as well?

**Review by liliankasem — APPROVED** [review]

File: test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SpecializationE2ETests.cs (hunk 1 above)
Comment (brettsam) [reply test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SpecializationE2ETests.cs:247]: Trying to think through some but can't come up with any that seem worthwhile. We're relying on the `Options` infrastructure to handle this (now that it's properly wired up) and the usage of `Lazy` so I think most of the corner cases are handled by those things.

Any ideas you had in mind?

Comment (liliankasem) [reply test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SpecializationE2ETests.cs:247]: nope :D

**Review by kashimiz — APPROVED** [review]
Looks good. Confirmed Function-level auth works when https://github.com/Azure/azure-functions-host/commit/2ba05ad5df0d63b82865c95a23314a415b64fbaa is layered on top.
